config/initializers/version.rb

```diff
@@ -0,0 +1,26 @@
+# config/initializers/version.rb
+
+if Rails.env.development?
+    # we will hae to use both the `git` and `rugged` gem until
+    # rugged implaments `git describe`
+    g = Git.open(Rails.root)
+    repo = Rugged::Repository.discover(Rails.root)
+
+    version = {
+        version: g.describe,
+        branch: repo.head.name.split('/').last,
+        sha: repo.head.target_id
+    }
+
+  File.open('config/VERSION','w') do |f|
+     f.write(JSON.pretty_generate(version))
```

Comment:
I'm not quite sure this will work as intended. It actually might be completely impossible to add a version in this manner since adding the version to the hash itself will change what the hash should be. Also, any squashes or merges done on Github will all have the wrong SHA since rails is not run on those. Heroku used to provide a `SOURCE_VERSION` ENV but is seems they have removed it.